src/ui/public/vis_maps/kibana_map.js

```diff
@@ -0,0 +1,507 @@
+import { EventEmitter } from 'events';
```

Comment:
The snapping back to the map is better, but the data is empty when it happens:

![multiple-earths](https://cloud.githubusercontent.com/assets/627123/24312953/4b8754e2-10b1-11e7-8ac4-530dab4b72fc.gif)